### PR TITLE
[SYCL][DOC] USM Proposal Update

### PR DESCRIPTION
**Pull request by jbrodman**
Rev proposal based on feedback.

### CODE REVIEW

File: sycl/doc/extensions/usm/usm.adoc [hunk 1]
```diff
@@ -22,139 +24,269 @@ Unified Addressing guarantees that all devices will use a unified address space.
...
-  event sycl_memset(void* ptr, int value, size_t count);
+  event memset(void* ptr, int value, size_t count);
 };
 
 class queue {
  ...
  public:
   ...
-  event sycl_memset(void* ptr, int value, size_t count);
+  event memset(void* ptr, int value, size_t count);
```
[the remark this answers is not in the document]
Comment (jbrodman) [reply sycl/doc/extensions/usm/usm.adoc]: This needs to be contiguous byte-based like std::memset.

Iterators don't have that requirement, right?

File: sycl/doc/extensions/usm/usm.adoc [hunk 2]
```diff
@@ -22,139 +24,269 @@ Unified Addressing guarantees that all devices will use a unified address space.
...
-==== memcpy
+===== memcpy
 [source,cpp]
 ----
 class handler {
  ...
  public:
   ...
-  event sycl_memcpy(void* dest, const void* src, size_t count);
+  event memcpy(void* dest, const void* src, size_t count);
```
[the remark this answers is not in the document]
Comment (jbrodman) [reply sycl/doc/extensions/usm/usm.adoc]: This needs to be contiguous byte copy like std::memcpy.  

Iterators don't have that requirement, right?

[the remark this answers is not in the document]
Comment (jbrodman) [reply sycl/doc/extensions/usm/usm.adoc]: Also, I don't like adding a flavor of handler::copy for this, b/c those are based on objects that have their size baked in, hence good ol' std::memcpy style operations.

Note - we can't use std::memcpy with device pointers b/c we need to handle them specially.

File: sycl/doc/extensions/usm/usm.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (keryell) [reply sycl/doc/extensions/usm/usm.adoc]: `void *` is not an iterator anyway.
But of course if you prefer you can use `memset`.

File: sycl/doc/extensions/usm/usm.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (keryell) [reply sycl/doc/extensions/usm/usm.adoc]: As you want.
This was just to propose some C++-like naming.
But C-like naming is fine.
In both case anyway it not exactly the same as either the C or C++ equivalent.

[the remark this answers is not in the document]
Comment (jbrodman) [reply sycl/doc/extensions/usm/usm.adoc]: Yep.  The good news is that for shared/host allocations, regular std::fill/copy_n should work!

File: sycl/doc/extensions/usm/usm.adoc [hunk 3]
```diff
@@ -325,21 +372,17 @@ Parameters::
...
+==== Concurrent USM
+Concurrent USM contains all the utility functions of Explicit USM and Restricted USM.  It introduces a new function, `sycl::mem_advise`, that allows programmers to provide additional information to the underlying runtime about how different allocations are used.  
 
-==== prefetch
+===== Performance Hints
+===== prefetch
 In Concurrent USM, prefetch commands may be overlapped with kernel execution.
-
-==== sycl_mem_advise
+===== mem_advise
```
Comment (bader) [inline sycl/doc/extensions/usm/usm.adoc]: Maybe it's just a GitHub artifact, but I see following in the rendered version:

In Concurrent USM, prefetch commands may be overlapped with kernel execution. ===== mem_advise

File: sycl/doc/extensions/usm/usm.adoc [hunk 4]
```diff
@@ -22,139 +24,269 @@ Unified Addressing guarantees that all devices will use a unified address space.
...
+
+|`shared`
+|Allocations in shared memory that are accessible by both host and device.  
 |===
 
-[cols="6*^",options="header",stripes=none]
+[cols="6*^",options="header", stripes=none]
 |===
-|Allocation Type |Initial Location |Accessible By | |Migratable To |
+|Allocation Type |Initial Location |Accessibly By | |Migratable To |
```
[the remark this answers is not in the document]
Comment (jbrodman) [reply sycl/doc/extensions/usm/usm.adoc]: Fixed.

File: sycl/doc/extensions/usm/usm.adoc (hunk 3 above)
Comment (jbrodman) [reply sycl/doc/extensions/usm/usm.adoc]: Good catch.

**Comment by bader** [comment]
http://ec2-52-15-243-124.us-east-2.compute.amazonaws.com:8010/#/builders/18/builds/16

@jbrodman, could you merge `sycl` branch to your branch, please?
We recently added Windows systems to CI and it seems it relies on some recent changes in the `sycl` branch.
@vladimirlaz, am I right?

**Comment by jbrodman** [comment]
Done.

**Comment by bader** [comment]
okay. It didn't help.
@vladimirlaz, @DoyleLi, @tfzhu could you fix sycl-win-x64-pr ASAP.

**Comment by DoyleLi** [comment]
> okay. It didn't help.
> @vladimirlaz, @DoyleLi, @tfzhu could you fix sycl-win-x64-pr ASAP.

Hi Alexey, Vladimir has already merged the code for windows build in the PR:

https://github.com/intel/llvm/pull/463
[SYCL] Enable build and LIT tests on windows

You can update your code to make win-x64-pr pass

**Comment by bader** [comment]
> > okay. It didn't help.
> > @vladimirlaz, @DoyleLi, @tfzhu could you fix sycl-win-x64-pr ASAP.
> 
> Hi Alexey, Vladimir has already merged the code for windows build in the PR:
> 
> #463
> [SYCL] Enable build and LIT tests on windows
> 
> You can update your code to make win-x64-pr pass

@DoyleLi, could you clarify what do you mean by "update your code"? 
I think James merged this patch to this PR. Am I wrong?

**Comment by DoyleLi** [comment]
> > > okay. It didn't help.
> > > @vladimirlaz, @DoyleLi, @tfzhu could you fix sycl-win-x64-pr ASAP.
> > 
> > 
> > Hi Alexey, Vladimir has already merged the code for windows build in the PR:
> > #463
> > [SYCL] Enable build and LIT tests on windows
> > You can update your code to make win-x64-pr pass
> 
> @DoyleLi, could you clarify what do you mean by "update your code"?
> I think James merged this patch to this PR. Am I wrong?

I mean you can update the files under the folder "buildbot" in your own branch from "https://github.com/intel/llvm/pull/463":

buildbot/compile.py 
buildbot/configure.py
buildbot/dependency.py

**Comment by bader** [comment]
> I think James merged this patch to this PR. Am I wrong?

I was wrong. 
@jbrodman, could you merge #463 to you branch, please?
Your branch is currently "89 commits behind intel:sycl."

**Comment by jbrodman** [comment]
Updated.

**Pull request merged**